File: cdk-lambda-chaos/lib/serverless-api/python/lambda-functions/createItem.py
import os
import json
import boto3

dynamodb = boto3.resource('dynamodb')

def handler(event, context):
    # Retrieve table name and primary key from environment variables
    table_name = os.environ.get('TABLE_NAME')
    primary_key = os.environ.get('PRIMARY_KEY')
    
    if not table_name or not primary_key:
        raise ValueError('TABLE_NAME and PRIMARY_KEY environment variables must be set')
    
    table = dynamodb.Table(table_name)
    
    # Assume the JSON payload is in the 'body' key of the event
    try:
        payload = json.loads(event['body'])
    except (KeyError, json.JSONDecodeError):
        return {
            'statusCode': 400,
            'body': json.dumps({'error': 'Invalid JSON payload'})
        }
    
    # Check if the primary key is present in the payload
    if primary_key not in payload:
        return {
            'statusCode': 400,
            'body': json.dumps({'error': f'Missing primary key: {primary_key}'})
        }
    
    # Write the payload to DynamoDB
    try:
        table.put_item(Item=payload)
    except Exception as e:
        return {
            'statusCode': 500,
            'body': json.dumps({'error': str(e)})
        }
    
    return {
        'statusCode': 200,
        'body': json.dumps({'message': 'Item saved successfully'})
    }

File: cdk-lambda-chaos/lib/serverless-api/typescript/lambda-functions/createItem.mjs
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import {
  DynamoDBDocumentClient,
  ScanCommand,
  PutCommand,
  GetCommand,
  DeleteCommand,
} from "@aws-sdk/lib-dynamodb";

const client = new DynamoDBClient({});
const dynamo = DynamoDBDocumentClient.from(client);

const TABLE_NAME = process.env.TABLE_NAME || 'Orders';
const PRIMARY_KEY = process.env.PRIMARY_KEY || 'Id';

export const handler = async (event, context) => {
    let body;
    let statusCode = 200;
    const headers = {
      "Content-Type": "application/json",
    };
    const item = typeof event.body == 'object' ? event.body : JSON.parse(event.body);
    if(!item[PRIMARY_KEY]){
        item[PRIMARY_KEY] = "001";
    }    
    try {
        body = await dynamo.send(
            new PutCommand({
              TableName: TABLE_NAME,
              Item: item
            })
          );
    } catch (err) {
        statusCode = 400;
        body = err.message;
      } finally {
        body = JSON.stringify(body);
      }
    
      return {
        statusCode,
        body,
        headers,
      };
};